File: apps/restaurant-user-frontend/src/services/orderService.js
import orderApi from "../api/orderApi";

export const createOrder = async (orderData) => {
  const response = await orderApi.post("/orders", orderData);
  return response.data;
};

export const getOrderById = async (id) => {
  const response = await orderApi.get(`/orders/${id}`);
  return response.data;
};

export const getAllOrders = async () => {
  const response = await orderApi.get("/orders");
  return response.data;
};

export const getOrdersByCustomerId = async (customerId) => {
  const response = await orderApi.get(`/orders/customer/${customerId}`);
  return response.data;
};

export const updateOrderStatus = async (id, status) => {
  const response = await orderApi.put(`/orders/${id}/status`, { status });
  return response.data;
};

export const deleteOrder = async (id) => {
  const response = await orderApi.delete(`/orders/${id}`);
  return response.data;
};

File: apps/order-service/order/src/main/java/com/example/order/service/OrderService.java
package com.example.order.service;

import com.example.order.dto.CreateOrderRequest;
import com.example.order.model.Order;
import com.example.order.model.OrderStatus;

import java.util.List;

public interface OrderService {

    Order createOrder(CreateOrderRequest request);

    Order getOrderById(String id);

    List<Order> getAllOrders();

    List<Order> getOrdersByCustomerId(String customerId);

    Order updateOrderStatus(String id, OrderStatus status);

    void deleteOrder(String id);
}